File: selenium-framework/src/test/java/com/selenium/testng/web/base/BasePage.java
package com.selenium.testng.web.base;

import org.openqa.selenium.WebDriver;

import com.selenium.testng.utils.ElementActions;

public class BasePage {

    protected WebDriver driver;
    protected ElementActions actions;

    public BasePage(WebDriver driver) {

        this.driver = driver;
        this.actions = new ElementActions(driver);
    }
    
    protected String getPageTitle() {
        return driver.getTitle();
    }

    protected String getCurrentUrl() {
        return driver.getCurrentUrl();
    }

    protected void refreshPage() {
        driver.navigate().refresh();
    }

    protected void navigateBack() {
        driver.navigate().back();
    }

    protected void navigateForward() {
        driver.navigate().forward();
    }
}

File: src/framework/web/base-page.ts
import type { Page } from '@playwright/test';
import { ElementActions } from '../utils/element-actions.js';

export class BasePage {
  protected readonly page: Page;
  protected readonly actions: ElementActions;

  constructor(page: Page) {
    this.page = page;
    this.actions = new ElementActions(page);
  }

  protected async getPageTitle(): Promise<string> {
    return this.page.title();
  }

  protected async getCurrentUrl(): Promise<string> {
    return this.page.url();
  }

  protected async refreshPage(): Promise<void> {
    await this.page.reload();
  }

  protected async navigateBack(): Promise<void> {
    await this.page.goBack();
  }

  protected async navigateForward(): Promise<void> {
    await this.page.goForward();
  }
}
